src/components/Navigation.js

```diff
@@ -0,0 +1,101 @@
+import styled from "styled-components";
+import { useRouter } from "next/router";
+
+export default function Navigation() {
+  const router = useRouter();
+  const { index } = router.query;
+  const currentPath = router.pathname;
+
+  function handleDownstairs() {
+    if (currentPath === "/boss-floor") {
+      router.push("/floor/2");
+    } else if (index === "1") {
+      return;
+    } else {
+      router.push(`/floor/${parseInt(index, 10) - 1}`);
+    }
+  }
+
+  function handleUpstairs() {
+    if (currentPath === "/boss-floor") {
+      return;
+    } else if (index === "2") {
+      router.push("/boss-floor");
+    } else {
+      router.push(`/floor/${parseInt(index, 10) + 1}`);
+    }
+  }
```

Comment:
To be honest, I'm not a big fan of imperative navigation (e.g. `router.push`). Buttons are for actions, links for navigation. 🫡 But  maybe that's a petty comment.